src/com/amazon/corretto/crypto/provider/AesCbcSpi.java

```diff
@@ -0,0 +1,704 @@
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES/CBC/PKCS7Padding".toLowerCase());
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES_128/CBC/PKCS7Padding".toLowerCase());
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES_192/CBC/PKCS7Padding".toLowerCase());
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES_256/CBC/PKCS7Padding".toLowerCase());
+    // PKCS5Padding with AES/CBC must be treated as PKCS7Padding. PKCS7Padding name is not
+    // recognized by SunJCE, but BouncyCastle supports PKCS7Padding as a valid name for the same
+    // padding.
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES/CBC/PKCS5Padding".toLowerCase());
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES_128/CBC/PKCS5Padding".toLowerCase());
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES_192/CBC/PKCS5Padding".toLowerCase());
+    AES_CBC_PKCS7_PADDING_NAMES.add("AES_256/CBC/PKCS5Padding".toLowerCase());
+  }
+
+  private static final byte[] EMPTY_ARRAY = new byte[0];
+  private static final int NO_PADDING = 0;
+  private static final int PKCS7_PADDING = 1;
+  private static final int BLOCK_SIZE_IN_BYTES = 128 / 8;
+  private static final int MODE_NOT_SET = -1;
+  private static final int ENC_MODE = 1;
+  private static final int DEC_MODE = 0;
+
+  private enum CipherState {
+    NEEDS_INITIALIZATION,
+    INITIALIZED,
+    UPDATED,
+  }
+
+  // State
+  private CipherState cipherState;
+  private final int padding;
+  // CBC processes data one block at a time. There are two scenarios where not all the input passed
+  // to engineUpdate is processed:
+  //     1. Input length is not a multiple of the block size,
+  //     2. Padding is enabled and cipher is configured for decryption.
+  // This variable keeps track of the unprocessed bytes.
+  private int unprocessedInput;
+  private int opMode;
+  private byte[] key;
+  private byte[] iv;
+  // nativeCtx is used to avoid memory leaks in case of multi-step operations or when the
+  // EVP_CIPHER_CTX needs to be preserved.
+  private NativeEvpCipherCtx nativeCtx;
+  // Determines if the EVP_CIPHER_CTX used should be released after doFinal or not. This is
+  // controlled by a system property.
+  private final boolean saveContext;
+
+  AesCbcSpi(final boolean paddingEnabled, final boolean saveContext) {
+    this.padding = paddingEnabled ? PKCS7_PADDING : NO_PADDING;
+    this.cipherState = CipherState.NEEDS_INITIALIZATION;
+    this.unprocessedInput = 0;
+    this.opMode = MODE_NOT_SET;
+    this.key = null;
+    this.iv = null;
+    this.nativeCtx = null;
+    this.saveContext = saveContext;
+  }
+
+  @Override
+  protected void engineSetMode(final String mode) throws NoSuchAlgorithmException {
+    // no op. One only needs to provide an implementation if the same Spi class instance can be used
+    // for different modes.
+  }
+
+  @Override
+  protected void engineSetPadding(final String padding) throws NoSuchPaddingException {
+    // no op. One only needs to provide an implementation if the same Spi class instance is used for
+    // different paddings.
```

Comment:
We do, but this method is not exposed to the user. The Cipher class uses it and since we only instantiate the Spi with the correct padding, the check here would be redundant. There are providers that don't set the padding in their Spi classes when it is instantiated. For those, it would be a good idea to override this method.

Reply:
I see, makes sense.